src/Commands/Plugin/UpdateCommand.php

```diff
@@ -23,6 +25,16 @@ class UpdateCommand extends PluginBaseCommand
      */
     public function update(array $plugins)
     {
+        // Check for minimum plugin command requirements.
```

Comment:
Instead of duplicating the messages et. al. for commandExists in each command, you could instead further factor this into a checkRequirements method that does the command checking and exception throwing. Also, I think there's little value in allowing someone to install but not update w/out installing git. You could just check for git and composer for all commands. Keep it simple.